src/AutoRest.CSharp/LowLevel/Output/LowLevelClient.cs

```diff
@@ -121,33 +122,96 @@ public static IEnumerable<LowLevelClientMethod> BuildMethods(RestClientBuilder b
             }
         }
 
+        private Parameter CreateOptionsParameter(bool isRequired)
+        {
+            var clientOptionsType = ClientOptions.Type.WithNullable(true);
+            return new Parameter("options", "The options for configuring the client.", clientOptionsType, isRequired ? null : Constant.NewInstanceOf(clientOptionsType), false);
+        }
+
         private IEnumerable<ConstructorSignature> BuildPublicConstructors()
         {
             if (!_hasPublicConstructors)
             {
                 yield break;
             }
 
-            var clientOptionsType = ClientOptions.Type.WithNullable(true);
-            var clientOptionsParameter = new Parameter("options", "The options for configuring the client.", clientOptionsType, Constant.NewInstanceOf(clientOptionsType), false);
-
             if (Fields.CredentialFields.Count == 0)
             {
-                yield return BuildPublicConstructor(null, clientOptionsParameter);
+                foreach (var constructor in BuildPublicConstructor(null))
```

Comment:
We probably need to split constructors with bodies and without bodies into two separate properties and generate them separately: first, produce all constructors with bodies. Then, use them to generate constructors without bodies that either call other constructors or just protected constructor for mocking.

Reply:
@AlexanderSher I though about this before to make MinimalPublicConstructors a new property. But the parameter list of these constructors are so related that I need to add more private fields for these parameters, otherwise I need to calculate the same parameter list every time I produce the constructors. Let me know your preference.

Reply:
Common part of parameters can be determined in constructor before creation of constructors starts, e.g.:
```cs
if (!IsSubClient)
{
    var requiredParameters = parameters.Where(p => p.IsRequired).ToArray();
    var optionalParameters = parameters
        .Where(p => !p.IsRequired && !p.IsApiVersionParameter)
        .Append(CreateClientOptionsParameter())
        .ToArray();

    PrimaryConstructors = BuildPrimaryConstructors(requiredParameters, optionalParameters).ToArray();
    SecondaryConstructors = BuildSecondaryConstructors(requiredParameters, optionalParameters).ToArray();
}
else
{
    PrimaryConstructors = Array.Empty<ConstructorSignature>();
    SecondaryConstructors = new[]{ CreateMockingConstructor() };
}
```
where 
```cs
private Parameter CreateClientOptionsParameter()
{
    var clientOptionsType = ClientOptions.Type.WithNullable(true);
    return new Parameter("options", "The options for configuring the client.", clientOptionsType, Constant.NewInstanceOf(clientOptionsType), false);
}
```

Reply:
With this implementation, `BuildPrimaryConstructors` and `BuildSecondaryConstructors` will be something like this:
```cs
private IEnumerable<ConstructorSignature> BuildPrimaryConstructors(IReadOnlyList<Parameter> requiredParameters, IReadOnlyList<Parameter> optionalParameters)
{
    var optionalAsRequired = optionalParameters.Select(p => p with {DefaultValue = null}).ToArray();
    if (Fields.CredentialParameters.Count == 0)
    {
        yield return CreatePrimaryConstructor(requiredParameters.Concat(optionalAsRequired).ToArray());
    }
    else
    {
        foreach (var credentialParameter in Fields.CredentialParameters)
        {
            yield return CreatePrimaryConstructor(requiredParameters.Append(credentialParameter).Concat(optionalAsRequired).ToArray());
        }
    }
}

private IEnumerable<ConstructorSignature> BuildSecondaryConstructors(IReadOnlyList<Parameter> requiredParameters, IReadOnlyList<Parameter> optionalParameters)
{
    if (requiredParameters.Any() || Fields.CredentialParameters.Any())
    {
        yield return CreateMockingConstructor();
    }

    var optionalParametersArguments = optionalParameters
        .Select(p => p.Type.EqualsIgnoreNullable(typeof(Uri)) ? $"new {typeof(Uri):I}({p.DefaultValue!.Value.GetFormattable()})" : p.DefaultValue!.Value.GetFormattable())
        .ToArray();

    if (Fields.CredentialParameters.Count == 0)
    {
        yield return CreateSecondaryConstructor(requiredParameters, optionalParametersArguments);
    }
    else
    {
        foreach (var credentialParameter in Fields.CredentialParameters)
        {
            yield return CreateSecondaryConstructor(requiredParameters.Append(credentialParameter).ToArray(), optionalParametersArguments);
        }
    }
}

private ConstructorSignature CreateMockingConstructor()
    => new(Declaration.Name, $"Initializes a new instance of {Declaration.Name} for mocking.", Protected, Array.Empty<Parameter>());

private ConstructorSignature CreatePrimaryConstructor(IReadOnlyList<Parameter> parameters)
    => new(Declaration.Name, $"Initializes a new instance of {Declaration.Name}", Public, parameters);

private ConstructorSignature CreateSecondaryConstructor(IReadOnlyList<Parameter> parameters, FormattableString[] optionalParametersArguments)
{
    var arguments = parameters
        .Select<Parameter, FormattableString>(p => $"{p.Name}")
        .Concat(optionalParametersArguments)
        .ToArray();
    return new(Declaration.Name, $"Initializes a new instance of {Declaration.Name}", Public, parameters, new ConstructorInitializer(false, arguments));
}
```

Reply:
This implementation does not follow https://github.com/Azure/autorest.csharp/issues/2055#issuecomment-1076868591. Currently I still fix by following this guidance until we have a conclusion about the outcome.